src/blockchain/rpc.rs

```diff
@@ -197,132 +207,151 @@ impl Blockchain for RpcBlockchain {
-                for output in tx.output.iter() {
-                    if let Ok(Some((kind, index))) =
-                        db.get_path_from_script_pubkey(&output.script_pubkey)
-                    {
-                        if index > *indexes.get(&kind).unwrap() {
-                            indexes.insert(kind, index);
-                        }
-                        received += output.value;
-                    }
+        // list_transactions returns conflicting txs too.
+        // Filter out conflicting: confirmation < 0 and not in mempool.
+        let listed_txs = listed_txs
+            .iter()
+            .filter(|t| {
+                t.info.confirmations > 0 || self.client.get_mempool_entry(&t.info.txid).is_ok()
+            })
+            .collect::<Vec<_>>();
+
+        let batch_update = loop {
+            request = match request {
+                Request::Script(script_req) => {
+                    let scripts = script_req.request();
+
+                    let satisfier = scripts
+                        .map(|sk| {
+                            let related_txs = listed_txs
+                                .iter()
+                                .filter(|tx_result| {
+                                    // Filter out txs related to the script_pubkey
+                                    if let Some(address) = &tx_result.detail.address {
+                                        if address.script_pubkey() == *sk {
+                                            // If conftime data is available, cache them.
+                                            if let (Some(height), Some(timestamp)) = (
+                                                tx_result.info.blockheight,
+                                                tx_result.info.blocktime,
+                                            ) {
+                                                txid_conftime_map.insert(
+                                                    tx_result.info.txid,
+                                                    ConfirmationTime { height, timestamp },
+                                                );
+                                            }
+                                            true
+                                        } else {
+                                            // address doesn't match
+                                            false
+                                        }
+                                    } else {
+                                        // There is no address associated with this tx
+                                        false
+                                    }
+                                })
+                                .map(|tx_result| (tx_result.info.txid, tx_result.info.blockheight))
+                                .collect::<Vec<_>>();
+                            related_txs
+                        })
+                        .collect::<Vec<_>>();
+
```

Comment:
I launched `cargo test --features test-rpc -- test_sync_bump_fee_remove_change -- --nocapture`
with a `dbg!(&satisfier)` here and noticed the bumped tx is not filtered out (causing the test to fail)

Reply:
Yes. The reason its happening is, in case of a tx without change address, the `listtransactions()` will only have an entry with the recipient's address. Which is an out of wallet address for those two test failure cases, and thus not fetched by the current simple filter.

I think this has to be fixed first anyway. My current plan is to get the difference between `listtransactions()` and `related_txs` sets, and apply `prev_out` fetch on the ones in the set difference. Match with the address in their prev_outs, and add them to appropriate list in the satisfier. 

Hopefully that can be done with `get_transaction()`, because the sought after inputs should always be a "owned" utxo, so its tx should be found. Also this way it will reduce too many `get_transaction()` call, we only fetch for tx with "owned inputs + not owned outputs". "owned output" txs will have already been filtered in the first address match try.

I hope this will work and at least will fix the test failures.